cxx4opencl/address_spaces.txt

```diff
@@ -98,6 +82,83 @@ it is legal). For example, a method can be implemented to exploit memory
 access coalescing for segments with memory bank. This not only applies
 to regular member functions but to constructors and destructors too.
 
+Address spaces are not deduced for:
+
+  * non-pointer/non-reference template parameters or any dependent types
+    except for template specializations.
+  * non-pointer/non-reference class members except for static data members
+    that are deduced to `__global` address space.
+  * non-pointer/non-reference alias declarations.
+  * decltype expressions.
+
+[source,c]
+----------
+template <typename T>
+void foo() {
+  T m; // address space of 'm' will be known at template instantiation time.
+  T * ptr; // 'ptr' points to generic address space object.
+  T & ref = ...; // 'ref' references an object in generic address space.
+};
+
+template <int N>
```

Comment:
Is the fact that this struct is templated important to the deduction rules?  N is never relevant.

Reply:
I think the aim of this example is to show that the address space deduction is performed normally if the type of variable doesn't depend on the type of a template parameter. In this case it is using a value of template parameter and not a type.

Reply:
Okay, thanks.